File: Personal/消息转发/消息转发/BaseViewController.m
//
//  BaseViewController.m
//  消息转发
//
//  Created by vcyber on 17/9/4.
//  Copyright © 2017年 vcyber. All rights reserved.
//

#import "BaseViewController.h"
#import "AppDelegate.h"

@interface BaseViewController ()

@end

@implementation BaseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (void)setBackgroundColor {
    self.view.backgroundColor = [UIColor redColor];
}

- (NSMethodSignature *)methodSignatureForSelector:(SEL)aSelector {
    if (aSelector == NSSelectorFromString(@"changeColor")) {
        return [NSMethodSignature signatureWithObjCTypes:"v@:"];
    }
    return [super methodSignatureForSelector:aSelector];
}

- (void)forwardInvocation:(NSInvocation *)anInvocation {
    if (anInvocation.selector == NSSelectorFromString(@"changeColor")) {
        AppDelegate *delgate = (AppDelegate *)[UIApplication sharedApplication].delegate;
        for (BaseViewController *c in delgate.targets) {
            [c setBackgroundColor];
        }
    }else {
        [super forwardInvocation:anInvocation];
    }
}


/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Personal/Swift/ComicAndPicture/ComicAndPicture/ComicAndPicture/BaseViewController.swift
//
//  BaseViewController.swift
//  ComicAndPicture
//
//  Created by vcyber on 16/11/2.
//  Copyright © 2016年 vcyber. All rights reserved.
//

import UIKit

class BaseViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.view.backgroundColor = UIColor.white
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.navigationController?.pushViewController(BaseViewController(), animated: true);
    }
    
    
    func addBackItem() -> Void {
        let imageView: UIImageView = UIImageView(frame: CGRect(x: 0, y: 0, width: 30, height: 30));
        imageView.image = UIImage(named: "back");
        imageView.isUserInteractionEnabled = true;
        let tap:UITapGestureRecognizer = UITapGestureRecognizer(target: self, action:#selector(back))
        imageView.addGestureRecognizer(tap);
        self.navigationItem.leftBarButtonItem = UIBarButtonItem.init(customView: imageView)
    }
    
    func back() {
        self.navigationController!.popViewController(animated: true);
    }
    
  
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
